package_test.go

```diff
@@ -581,6 +581,80 @@ func (s *PackageSuite) TestAllErrors(c *C) {
 	}
 }
 
+func (s *PackageSuite) TestTransactions(c *C) {
+	dropTables, sqldb, err := personAndAddressDB()
+	if err != nil {
+		c.Fatal(err)
+	}
+
+	selectStmt := sqlair.MustPrepare("SELECT &Person.* FROM person WHERE address_id = $Person.address_id", Person{})
+	insertStmt := sqlair.MustPrepare("INSERT INTO person VALUES ( $Person.name, $Person.id, $Person.address_id, 'fred@email.com');", Person{})
+	var derek = Person{ID: 85, Fullname: "Derek", PostalCode: 8000}
+
+	db := sqlair.NewDB(sqldb)
+	tx, err := db.Begin(nil, nil)
+	if err != nil {
+		c.Fatal(err)
+	}
+	// Insert derek then rollback.
+	q := tx.Query(nil, insertStmt, &derek)
+	// TODO: replace this with q.Run() or equivlent.
+	iter := q.Iter()
+	iter.Next()
+	err = iter.Close()
+	if err != nil {
+		c.Fatal(err)
+	}
+	err = tx.Rollback()
+	if err != nil {
+		c.Fatal(err)
+	}
+
+	// Check derek isnt in db, insert derek; commit.
+	tx, err = db.Begin(nil, nil)
+	if err != nil {
+		c.Fatal(err)
+	}
+	var derekCheck = Person{}
+	err = tx.Query(nil, selectStmt, &derek).One(&derekCheck)
+	if !errors.Is(err, sqlair.ErrNoRows) {
+		c.Fatalf("test failed, got err %s, expected %s", err, sqlair.ErrNoRows)
```

Comment:
No need to say "test failed" on test failures. Not sure if we have other similar cases?

In this particular case, though:
```
c.Assert(err, Equals, sql.ErrNoRows)
```

Reply:
Changed.